sympy/concrete/summations.py

```diff
@@ -401,6 +403,10 @@ def is_convergent(self):
             lower_limit = -upper_limit
             upper_limit = S.Infinity
 
+        sym_ = Dummy(sym.name, integer=True, positive=True)
+        sequence_term = sequence_term.xreplace({sym: sym_})
+        sym = sym_
+
```

Comment:
Shouldn't the assumption of an integer be implemented in `ExprWithIntLimits`?

Reply:
Actually, I think this change would not be needed if the tests used `limit_seq` instead of `limit`, because `limit_seq` already carries out this substitution of an integer dummy. Try replacing `limit` with `limit_seq` in this method, it will probably improve the results without need for dummy replacement.

Reply:
But, a problem will arise then.
@normalhuman , If we replace `limit` with `limit_seq` , in some tests, we will be getting `TypeError: can't convert oo to int`. Now, I tried to look into that, but couldn't find any plausible solution. I think it will be better to keep the original change.